doc/guides/abi-stability.md

```diff
@@ -0,0 +1,107 @@
+# ABI Stability
+
+## Introduction
+An Application Binary Interface (ABI) is the compiled version of an Application
+Programming Interface (API). That is, the headers files describing the classes,
+functions, data structures, enumerations, and constants used to enable an
+application to perform a desired task correspond by way of compilation to a set
+of addresses and expected parameter values and memory structure sizes and
+layouts with which the provider of the ABI was compiled.
+
+The application using the ABI must be compiled such that the available
+addresses, expected parameter values, and memory structure sizes and layouts
+agree with those with which the ABI provider was compiled. This is usually
+accomplished by compiling against the headers provided by the ABI provider.
+
+Since the ABI provider and the ABI user may be compiled at different times with
+different versions of the compiler, a portion of the responsibility for
+ensuring ABI compatibility lies with the compiler. Different versions of the
+compiler, perhaps provided by different vendors, must all produce the same ABI
+from a header file with a certain content, and must produce code for the
+application using the ABI that accesses the API described in a given header
+according to the conventions of the ABI resulting from the description in the
+header. Modern compilers have a fairly good track record of not breaking the
+ABI compatibility of the applications they compile.
+
+The remaining responsibility for ensuring ABI compatibility lies with the team
+maintaining the header files which provide the API that results, upon
+compilation, in the ABI that is to remain stable. Changes to the header files
+can be made, but the nature of the changes has to be closely tracked to ensure
+that, upon compilation, the ABI does not change in a way that will render
+existing users of the ABI incompatible with the new version.
+
+## ABI Stability in Node.js
+Node.js provides header files maintained by several independent teams. For
```

Comment:
I would link to the N-API and native-addon docs here perhaps?

Reply:
You mean https://nodejs.org/docs/latest/api/addons.html and https://nodejs.org/docs/latest/api/n-api.html? I don't quite understand why I would link to the docs here?